packages/lodestar/src/util/retry.ts

```diff
@@ -28,7 +28,7 @@ export async function retry<A>(fn: (attempt: number) => A | Promise<A>, opts?: I
   const shouldRetry = opts?.shouldRetry;
 
   let lastError: Error = Error("RetryError");
-  for (let i = 1; i <= maxRetries; i++) {
+  for (let i = 0; i < maxRetries; i++) {
```

Comment:
Hi @g11tech can you help look into this? Given you made the modifications, it is probably faster to share any insights you had explaining the reason for the change. As far as I can see, everything still looks fine.

Reply:
hey @dadepo  I think i did it to not count first attempt as retry, you can revert it and may be correct a test that might fail because of retry comparison.